src/main/scala/org/apache/spark/sql/hive/llap/LlapContext.scala

```diff
@@ -124,6 +122,44 @@ class LlapContext(sc: SparkContext,
       Seq.empty
     }
   }
+
+  override protected[sql] def executePlan(plan: LogicalPlan): super.QueryExecution = {
+    new this.QueryExecution(plan)
+    super.executePlan(plan)
+  }
+
+  private[sql] class QueryExecution(logicalPlan: LogicalPlan)
```

Comment:
I think private would be ok, since this is also defined as private in the LlapContext